Library/ViewModels/RewardAddOnSelectionViewModel.swift

```diff
@@ -256,15 +256,55 @@ public final class RewardAddOnSelectionViewModel: RewardAddOnSelectionViewModelT
     .map(PledgeViewData.init)
     .takeWhen(self.continueButtonTappedProperty.signal)
 
+    let defaultShippingTotal = Signal.zip(project, baseReward)
+      .map { project, baseReward -> Double in
+        guard baseReward.shipping.enabled, let backing = project.personalization.backing else { return 0.0 }
+
+        return backing.shippingAmount.flatMap(Double.init) ?? 0.0
+      }
+
+    let calculatedShippingTotal = Signal.combineLatest(
+      shippingRule.skipNil(),
+      allRewards,
+      selectedQuantities
+    )
+    .map { shippingRule, rewards, selectedQuantities -> Double in
+      rewards.reduce(0.0) { total, reward in
+        guard reward.shipping.enabled else { return total }
+
+        let shippingCostForReward = reward.shippingRule(matching: shippingRule)?.cost ?? 0
+
+        let totalShippingForReward = shippingCostForReward
+          .multiplyingCurrency(Double(selectedQuantities[reward.id] ?? 0))
+
+        return total.addingCurrency(totalShippingForReward)
+      }
+    }
+
+    let allRewardsShippingTotal = Signal.merge(
+      defaultShippingTotal,
+      calculatedShippingTotal
+    )
+
     // MARK: - Tracking
 
-    Signal.zip(project, baseReward, refTag)
+    Signal.zip(project, baseReward, selectedRewards, refTag, configData, allRewardsShippingTotal)
       .take(first: 1)
       .observeForUI()
-      .observeValues { project, baseReward, refTag in
+      .observeValues { project, baseReward, selectedRewards, refTag, configData, rewardsShippingTotal in
         AppEnvironment.current.ksrAnalytics.trackAddOnsPageViewed(
           project: project,
           reward: baseReward,
+          checkoutData: checkoutProperties(
+            from: project,
+            baseReward: baseReward,
+            rewards: selectedRewards,
+            selectedQuantities: configData.selectedQuantities,
+            additionalPledgeAmount: 0,
```

Comment:
Depending on if Laurie wants this event to handle an edited pledge this may require you to use the existing `Backing` (project.personalization.backing) instead of hard-coding 0.

Reply:
This has updated as well.